services/galley/src/Galley/API/Swagger.hs

```diff
@@ -0,0 +1,357 @@
+          , ("public_key", Inline (toSchema (Proxy @(ServiceKeyPEM))))
+          ]
+
+        example_ :: Maybe Value
+        example_ = Just . toJSON
+                 $ ViewLegalHoldServiceInfo (Id tid) lhuri fpr tok key
+          where
+            tok = ServiceToken "blablabla"
+            Just key = fromByteString . encodeUtf8 $ Text.unlines $
+                [ "-----BEGIN PUBLIC KEY-----"
+                , "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu+Kg/PHHU3atXrUbKnw0"
+                , "G06FliXcNt3lMwl2os5twEDcPPFw/feGiAKymxp+7JqZDrseS5D9THGrW+OQRIPH"
+                , "WvUBdiLfGrZqJO223DB6D8K2Su/odmnjZJ2z23rhXoEArTplu+Dg9K+c2LVeXTKV"
+                , "VPOaOzgtAB21XKRiQ4ermqgi3/njr03rXyq/qNkuNd6tNcg+HAfGxfGvvCSYBfiS"
+                , "bUKr/BeArYRcjzr/h5m1In6fG/if9GEI6m8dxHT9JbY53wiksowy6ajCuqskIFg8"
+                , "7X883H+LA/d6X5CTiPv1VMxXdBUiGPuC9IT/6CNQ1/LFt0P37ax58+LGYlaFo7la"
+                , "nQIDAQAB"
+                , "-----END PUBLIC KEY-----"
+                ]
+            Just tid = fromText "7fff70c6-7b9c-11e9-9fbd-f3cc32e6bbec"
+            Right lhuri = mkHttpsUrl [uri|https://example.com/|]
+            fpr = Fingerprint "\138\140\183\EM\226#\129\EOTl\161\183\246\DLE\161\142\220\239&\171\241h|\\GF\172\180O\129\DC1!\159"
+
+instance ToSchema LegalHoldTeamConfig where
+    declareNamedSchema = genericDeclareNamedSchema opts
+      where
+        opts = defaultSchemaOptions
+          { fieldLabelModifier = \case
+              "legalHoldTeamConfigStatus" -> "status"
+          }
+
+instance ToSchema LegalHoldStatus where
+    declareNamedSchema = tweak . genericDeclareNamedSchema opts
+      where
+        opts = defaultSchemaOptions
+          { constructorTagModifier = \case
+              "LegalHoldDisabled" -> "disabled"
+              "LegalHoldEnabled"  -> "enabled"
+          }
+
+        tweak = fmap $ schema . description ?~ descr
+          where
+            descr = "determines whether admins of a team " <>
+                    "are allowed to enable LH for their users."
+
+instance ToSchema RequestNewLegalHoldClient where
+    declareNamedSchema = genericDeclareNamedSchema opts
+      where
+        opts = defaultSchemaOptions
+          { fieldLabelModifier = \case
+              "userId" -> "user_id"
+              "teamId" -> "team_id"
+          }
+
+instance ToSchema NewLegalHoldClient where
+    declareNamedSchema = genericDeclareNamedSchema opts
+      where
+        opts = defaultSchemaOptions
+          { fieldLabelModifier = \case
+              "newLegalHoldClientPrekeys"     -> "prekeys"
+              "newLegalHoldClientLastKey"     -> "last_prekey"
+          }
+
+-- | this type is only introduce locally here to generate the schema for 'UserLegalHoldStatusResponse'.
+--   TODO: Use SimpleClientId in the UserLegalHoldStatusResponse datatype or re-use PubClient datatype
+--         which would greatly simplify these instances
```

Comment:
@fisx feel free to refactor this before unwipping. Also, the naming could be different 🙈